fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -516,6 +518,27 @@ public String getTableName() throws MetaNotFoundException {
         }
     }
 
+    // This method should be used by caller with other locks, like modifyJob()
+    public String getTableNameWithTimeout() throws UserException {
+        Database database = GlobalStateMgr.getCurrentState().getDb(dbId);
+        if (database == null) {
+            throw new MetaNotFoundException("Database " + dbId + "has been deleted");
+        }
+        if (!database.tryReadLock(GET_DB_LOCK_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+            throw new UserException("get database read lock timeout, database="
+                    + database.getFullName() + ", timeoutMillis=" + GET_DB_LOCK_TIMEOUT_MS);
+        }
```

Comment:
There is no need to hold db lock here. database.getTable is thread safe

Reply:
It's true, I missed the code [here](https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/catalog/Database.java#L146).

Reply:
Remove read lock to fix this issue, please check it again. Thank you!

Reply:
> ![image](https://user-images.githubusercontent.com/107105845/279141648-8d95664e-6be4-49dc-a299-9071c689f8e7.png) approved, but you need to fix the coverage issue before it's merged.

No code added, just removed the lock acquire code in the same function. Any suggestions?

Reply:
like setting up 2 threads in ut to simulate the dead lock situation and check no dead lock?